deploy.sh

```diff
@@ -158,15 +170,41 @@ export ELASTICSEARCH_INDEX_NAME_PREFIX="mc_search-*"
 export TERMFIELDS="article_title,text_content"
 export TERMAGGRS="top,significant,rare"
 export API_PORT
+export API_REPLICAS
 export UI_PORT
 export ESHOSTS
 export SENTRY_DSN
 export SENTRY_ENVIRONMENT
 export IMAGE_TAG
+export NEWS_SEARCH_UI_TITLE
+
+if $USE_LATEST_IMAGE; then
+    echo "Building Docker images..."
+    docker compose build
+    STATUS=$?
+    if [ $STATUS != 0 ]; then
+        echo "docker compose build failed: $STATUS" >&2
+        exit 1
+    fi
+fi
+
+# Deploy services using Docker Swarm
+docker stack config -c "$(pwd)/$DOCKER_COMPOSE_FILE" > processed.swarm.yml
```

Comment:
Comment (deploy) doesn't match code (config)

Reply:
I suggest changing the comment to be something like "Validate $DOCKER_COMPOSE_FILE"

Does this check/expand references to environment variables?

story-indexer deploy.sh calls this "docker-compose.yml.dump" (shorter than ".expanded").  Why not be consistent?  Adding to the end of the filename means the files will appear together in "ls" output and both can be grep'ed with less typing.

P.S.
Is $(pwd) really needed????  The story-indexer deploy script didn't need/use an absolute path here.

Reply:
Yes `docker stack config` does the syntax validation & expands the env variables hence we can `stack deploy` with processed yml file